### PR TITLE
Update CODEOWNERS 

### PR DESCRIPTION
**Pull request by bossinc**
Partner datasources taking over opentsdb, graphite, and influxdb

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -152,7 +152,7 @@
 /pkg/tests/apis/ @grafana/grafana-app-platform-squad
 /pkg/tests/api/correlations/ @grafana/explore-squad
 /pkg/tsdb/grafanads/ @grafana/grafana-backend-group
-/pkg/tsdb/opentsdb/ @grafana/grafana-backend-group
+/pkg/tsdb/opentsdb/ @grafana/partner-datasources
```
Comment (bossinc) [inline .github/CODEOWNERS:155]: I am unsure if this one should be updated because it was not previously owned by `observability-metrics`.

Comment (itsmylife) [reply .github/CODEOWNERS:155]: It was owned by us but apparently, that part wasn't updated.

**Review by itsmylife — APPROVED** [review]
🚀

**Review by bohandley — COMMENTED** [review]
🎉 🎉 🎉

**Pull request merged**